packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -625,37 +640,57 @@ class _RootBuilder extends dart_ast_visitor.RecursiveAstVisitor<Object?> {
         );
       }
     } else {
-      _currentClass = Class(name: node.name.name, fields: <Field>[]);
+      _currentClass = Class(name: node.name.name, fields: <NamedType>[]);
     }
 
     node.visitChildren(this);
     return null;
   }
 
+  NamedType formalParameterToField(dart_ast.FormalParameter parameter) {
+    final dart_ast.TypeName typeName = parameter.childEntities.firstWhere(
+        (dart_ast_syntactic_entity.SyntacticEntity e) =>
+            e is dart_ast.TypeName) as dart_ast.TypeName;
+    final String argType = typeName.name.name;
```

Comment:
Separately but semi-related `argType` is a pretty confusing name for the local given that there's `typeArguments` as a different concept in the same method. How about just `name`?

Reply:
`name` could easily be confused to mean the user defined name for the argument, not the name of the type.  `dataType: argType` makes more sense than `dataType: name`.  What we decide with the previous comment will inform what we do here.

Reply:
named `artTypeBaseName`